src/cryptography/hazmat/backends/openssl/ciphers.py

```diff
@@ -110,6 +112,17 @@ def update(self, data):
         if len(data) == 0 and not isinstance(self._mode, modes.GCM):
             return b""
 
+        self._bytes_processed += len(data)
+        if (
+            isinstance(self._mode, modes.GCM) and
```

Comment:
#1844 and #1847 were more generic solutions than this, but they ended up being higher complexity than we wanted. This issue has now been outstanding for 3 months, so I'm a bit frustrated that, despite it being a user bug report, we still don't have an answer. :(